src/main/java/org/apache/maven/plugins/assembly/io/FileLocation.java

```diff
@@ -0,0 +1,175 @@
+package org.apache.maven.plugins.assembly.io;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+
+
+/**
+ * file location implementation.
+ *
+ */
+class FileLocation
+    implements Location
+{
+
+    private File file;
+    private FileChannel channel;
+    private final String specification;
+    private FileInputStream stream;
+
+    /**
+     * @param file {@link File}
+     * @param specification spec
+     */
+    FileLocation( File file, String specification )
+    {
+        this.file = file;
+        this.specification = specification;
+    }
+
+    /**
+     * @param specification spec
+     */
+    FileLocation( String specification )
+    {
+        this.specification = specification;
+    }
+
+    /** {@inheritDoc} */
+    public void close()
+    {
+        if ( ( channel != null ) && channel.isOpen() )
+        {
+            try
+            {
+                channel.close();
+            }
+            catch ( IOException e )
+            {
+                //swallow it.
```

Comment:
Why do we swallow the exception? Isn't this bad practice?

Reply:
In general, yes. Exceptions are often swallowed when closing a resource though, under the assumption that nothing else important will happen with it.

Also, note that most of this code is just moved over from maven-shared-io. It's not actually new. I just wanted eliminate the dependencies on the much larger parts of maven-sahred-io we're not using here.